src/lib/stores/walletStore.js

```diff
@@ -95,19 +121,48 @@ function createWalletStore() {
          */
         sign: async ({ transactionXDR, network, pincode }) => {
             try {
-                const keyManager = setupKeyManager()
-                let signedTransaction = await keyManager.signTransaction({
+             
+                const { keyId, publicKey } = get(walletStore);
+                
+                if (keyId === publicKey) {
+                    const networkType = network.includes('Test') ? 
+                    WalletNetwork.TESTNET : 
+                    WalletNetwork.PUBLIC;
+    
+                const kit = new StellarWalletsKit({
+                    network: networkType,  // Use enum here
```

Comment:
like i mentioned before, it is indeed safe to use the `network` parameter that gets passed into the `sign(...)` function. You can also get rid of the ternary check to declare `networkType`, too.
```suggestion
                    network: network
```

Reply:
Okay I get you ser @ElliotFriend, I have added it